### PR TITLE
Veoss 629 - Amend returns confirmation for returns with corrections

### PR DESCRIPTION
**Pull request by warrenskiwork1979**

### CODE REVIEW

File: app/controllers/YourAccountController.scala [hunk 1]
```diff
@@ -106,6 +98,28 @@ class YourAccountController @Inject()(
...
           throw new Exception(error.toString)
       }
+  }
+
+  private def getFilteredPeriodsWithOutstandingAmounts(vatReturnsWithFinancialData: Seq[VatReturnWithFinancialData]) = {
+    financialDataService
+      .filterIfPaymentIsOutstanding(vatReturnsWithFinancialData)
+      .map(vatReturnWithFinancialData =>
+        vatReturnWithFinancialData.vatOwed match {
+          case Some(_) => vatReturnWithFinancialData //.copy(vatOwed = Some(vatOwed)) don't think we need this.
```
Comment (darbyo) [inline app/controllers/YourAccountController.scala:108]: Can we remove this comment,  if its not needed

File: app/controllers/YourAccountController.scala [hunk 2]
```diff
@@ -106,6 +98,28 @@ class YourAccountController @Inject()(
           logger.error(s"there was an error getting periods with outstanding amounts $error")
           throw new Exception(error.toString)
       }
+  }
+
+  private def getFilteredPeriodsWithOutstandingAmounts(vatReturnsWithFinancialData: Seq[VatReturnWithFinancialData]) = {
+    financialDataService
+      .filterIfPaymentIsOutstanding(vatReturnsWithFinancialData)
+      .map(vatReturnWithFinancialData =>
+        vatReturnWithFinancialData.vatOwed match {
```
Comment (darbyo) [inline app/controllers/YourAccountController.scala:107]: I wanted to check this part but never got time. In a scenario when we get back 'vatOwed' from vatReturnWithFinancialData, we just use that by default and don't add in the correction amount. Can we just double confirm that the vatOwed will have the correction amount included in it by default

File: app/services/VatReturnSalesService.scala [hunk 3]
```diff
@@ -53,4 +58,17 @@ class VatReturnSalesService @Inject()() {
     ).sum
   }
 
+  def getCorrectionAmount(periodWithCorrections: List[PeriodWithCorrections]): BigDecimal = {
+    periodWithCorrections.map { periodWithCorrection =>
+      periodWithCorrection.correctionsToCountry.map {
+        correctionToCountry =>
+          if (correctionToCountry.countryVatCorrection < 0) {
+            BigDecimal(0)
```
Comment (darbyo) [inline app/services/VatReturnSalesService.scala:66]: Why do we return 0 when countryVatCorrection < 0 ?

**Review by darbyo — CHANGES_REQUESTED** [review]

**Review by darbyo — APPROVED** [review]

File: app/controllers/YourAccountController.scala (hunk 2 above)
Comment (warrenskiwork1979) [reply app/controllers/YourAccountController.scala:106]: To Discuss with ETMP

**Pull request merged**